tests/test_trajectory.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+
+import unittest
+from ddt import ddt, data, unpack
+
+import numpy as np
+
+import broni
+from broni.shapes.primitives import Cuboid, Sphere
+
+
+@ddt
+class TestTrajectory(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_invalid(self):
+        with self.assertRaises(ValueError):
+            assert broni.Trajectory(
+                np.array([[0, 0], [0, 0]], dtype=float),
+                [2, 2],
+                coordinate_system="GSE"
+            )
+
+        with self.assertRaises(ValueError):
+            assert broni.Trajectory(
+                np.array([[0, 0, 0], [0, 0, 0]], dtype=float),
+                [1, 2, 3],
+                coordinate_system="GSE"
+            )
+
+    @data(
+        (np.array([[1, 1, 1]]), [Cuboid(0, 0, 0, 2, 2, 2)], np.array([[0, 0]])), # simple, one dot trajectory, objet-list
+        (np.array([[1, 1, 1]]), Cuboid(0, 0, 0, 2, 2, 2), np.array([[0, 0]])),  # single object (listified)
+
+        (np.array([[2, 2, 2]]), [Cuboid(0, 0, 0, 1, 1, 1)], np.array([])),  # all points are outside
+
+        (np.array([[2, 2, 2]]), [], np.array([])),  # no selection objects
+
+        (np.array([[-1, -1, -1], [1, 1, 1]]), [Cuboid(0, 0, 0, 2, 2, 2)], np.array([[1, 1]])),
+        (np.array([[-1, -1, -1], [1, 1, 1]]), [Cuboid(0, 0, 0, 2, 2, 2)], np.array([[1, 1]])),
+        (np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1]]), [Cuboid(0, 0, 0, 2, 2, 2)], np.array([[1, 2]])),
+
+        (np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1], [-1, -1, -1], [0, 0, 0], [1, 1, 1]]),
+         [Cuboid(0, 0, 0, 2, 2, 2)], np.array([[1, 2], [4, 5]])),
+
+        (np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1], [-1, -1, -1], [0, 0, 0], [1, 1, 1]]),
+         [Cuboid(0, 0, 0, 2, 2, 2), Sphere(1.5, 1.5, 1.5 , 1)], np.array([[2, 2], [5, 5]])),  # overlapping sphere and cuboid, logical_and
+    )
+    @unpack
+    def test_trajectory(self, trajectory, objects, expected):
```

Comment:
I don't think the test case name tells us what is tested. 
- test_trajectory seems to actually test the intervals
- even test_intervals is not really self explanatory...

considering the file is already named test_trajectory, maybe the test case should be something like  `test_intersections_with_user_objects`

Reply:
done